### DISCUSSION TITLE
Issues with Compilation Errors in CMAQv51+ Using KPP 3.1.1 and gfortran

### DISCUSSION THREAD

**Discussion by Caroramirezz** [KineticPreProcessor/KPP#109] [Q&A]
Hello KPP Community,

I'm currently working with a project originally written in KPP 2.2.3 and I've been encountering some issues when trying to compile the code with different versions of gfortran and KPP.

Setup:

Initial Setup: KPP 3.1.1 with the latest version of gfortran
Errors Encountered: When compiling, I encountered multiple "no IMPLICIT type" errors related to symbols in my Fortran code.
Attempted Fixes:
Downgraded gfortran to version 7.1: The initial error was replaced by a different set of issues related to the compiler not finding f951.
Switched KPP versions to 2.4 (oldest .tar file I could find): I continued to face similar issues, indicating that the problem might be related to either the gfortran version or incompatibilities between the KPP versions.
Given that the original code was written in KPP 2.2.3, I'm trying to determine the best combination of KPP and gfortran versions to use to successfully compile and run the code.

Has anyone faced similar issues? Any guidance on how to resolve this would be greatly appreciated.

Thanks in advance!

**Comment by RolfSander** [comment]
This problem seems to be identical to that shown here:

https://github.com/KineticPreProcessor/KPP/issues/110

**Comment by RolfSander** [comment]
When upgrading from an older KPP version to KPP 3, please read also
this:

https://kpp.readthedocs.io/en/stable/getting_started/00_revision_history.html#kpp-3-0-0
